# Logistic Regression

In [1]:
# yang di import
import pandas as pd
import numpy as np

In [3]:
df1 = pd.read_csv('calonpembeli.csv')
df1.describe()

ID         Usia       Status      Kelamin  Memiliki_Mobil  \
count  1000.000000  1000.000000  1000.000000  1000.000000      1000.00000   
mean    500.500000    43.532000     1.469000     0.481000         0.95200   
std     288.819436    12.672078     1.073402     0.499889         0.80146   
min       1.000000    24.000000     0.000000     0.000000         0.00000   
25%     250.750000    33.000000     1.000000     0.000000         0.00000   
50%     500.500000    43.000000     1.000000     0.000000         1.00000   
75%     750.250000    53.000000     2.000000     1.000000         2.00000   
max    1000.000000   164.000000     3.000000     1.000000         4.00000   

       Penghasilan   Beli_Mobil  
count   1000.00000  1000.000000  
mean     270.09000     0.633000  
std       95.23681     0.482228  
min       95.00000     0.000000  
25%      187.00000     0.000000  
50%      258.50000     1.000000  
75%      352.25000     1.000000  
max      490.00000     1.000000

di atas Usia maling tinggi adalah 164 tahun maka kita saring. hanya sampai 100 saja

In [4]:
df1 = df1[df1["Usia"] <= 100]

kita perlu melihat apakah ada data kosong (null) di dalamnya.

In [6]:
df1.isnull().sum()

ID                0
Usia              0
Status            0
Kelamin           0
Memiliki_Mobil    0
Penghasilan       0
Beli_Mobil        0
dtype: int64

Jika ada data yang NaN/Null maka kita Hilangkan dgn df1.dropna()
jika tidak ada maka langsung saja.

ingin melihat total pembeli sama yg tidak beli

In [7]:
df1["Beli_Mobil"].value_counts()

1    633
0    365
Name: Beli_Mobil, dtype: int64

0 adalah tidak jadi beli dan 1 benar benar pembeli mobil